pkg/systemd/quadlet/quadlet.go

```diff
@@ -1793,8 +1817,15 @@ func addNetworks(quadletUnitFile *parser.UnitFile, groupName string, serviceUnit
 					return fmt.Errorf("requested Quadlet unit %s was not found", quadletNetworkName)
 				}
 
+				var resourceName string
+				if isContainerUnit {
+					resourceName = getContainerResourceName(quadletUnitFile, unitInfo)
```

Comment:
This seems wrong?
quadletUnitFile == file of the current unit that contains Network=xxx.container

Then getContainerResourceName() calls getContainerName() with quadletUnitFile which means it will extract the name from the current unit not the target unit AFAICT
Th only way this can work is vause the unitInfo cache is set which you are claiming doesn't work always so I don't see how this fixes anything? Am I overlooking something?